qutebrowser/browser/urlmarks.py

```diff
@@ -218,6 +215,24 @@ def quickmark_del(self, name):
         except KeyError:
             raise cmdexc.CommandError("Quickmark '{}' not found!".format(name))
 
+    def get_by_qurl(self, url):
+        """Look up a quickmark by QUrl, returning its name.
+
+        Takes O(n) time, where n is the number of quickmarks.
+        Use a name instead where possible.
+        """
+        if not url.isValid():
+            raise ValueError("Invalid URL: {}".format(url.errorString()))
+        urlstr = url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)
+
+        try:
+            index = list(self.marks.values()).index(urlstr)
+            key = list(self.marks.keys())[index]
```

Comment:
You can also build the inverse dict just once and have O(1) search until the bookmarks change. But as I said, the thing is that the inverse generally does not exist.
@mlochbaum: How does your method handle multiple marks for one URL? I guess it would remove only the first one of the items in arbitrary order.